gradient/commands/datasets.py

```diff
@@ -561,141 +561,139 @@ def update_status():
+                # 15MB chunksize
+                #
+                # Note also that AWS limits the max number of chunks
+                # in a multipart upload to 10000, so this setting
+                # currently enforces a hard limit on 150GB per file.
+                #
+                # We can dynamically assign a larger part size if needed,
+                # but for the majority of use cases we should be fine
+                # as-is
+                part_minsize = int(15e6)
+                dataset_id, _, version = dataset_version_id.partition(":")
+                mpu_url = f'/datasets/{dataset_id}/versions/{version}/s3/preSignedUrls'
+                api_client = http_client.API(
+                    api_url=config.CONFIG_HOST,
+                    api_key=self.api_key,
+                    ps_client_name=CLI_PS_CLIENT_NAME
+                )
+
+                mpu_create_res = api_client.post(
+                    url=mpu_url,
+                    json={
+                        'datasetId': dataset_id,
+                        'version': version,
+                        'calls': [{
+                            'method': 'createMultipartUpload',
+                            'params': {'Key': key}
+                        }]
+                    }
+                )
+
+                mpu_data = mpu_create_res.json()[0]['url']
+
+                parts = []
+                with open(path, 'rb') as f:
+                    # we +2 the number of parts since we're doing floor
+                    # division, which will cut off any trailing part
+                    # less than the part_minsize, AND we want to 1-index
+                    # our range to match what AWS expects for part
+                    # numbers
+                    for part in range(1, (size // part_minsize) + 2):
+                        presigned_url_res = api_client.post(
+                            url=mpu_url,
+                            json={
+                                'datasetId': dataset_id,
+                                'version': version,
+                                'calls': [{
+                                    'method': 'uploadPart',
+                                    'params': {
+                                        'Key': key,
+                                        'UploadId': mpu_data['UploadId'],
+                                        'PartNumber': part
+                                    }
+                                }]
+                            }
+                        )
+
+                        presigned_url = presigned_url_res.json()[0]['url']
+
+                        chunk = f.read(part_minsize)
+
+                        for attempt in range(0, 5):
+                            part_res = session.put(
+                                presigned_url,
+                                data=chunk,
+                                headers=headers,
+                                timeout=300)
```

Comment:
Increase timeout from 5 seconds to 5 minutes

Reply:
maybe this timeout can be pulled out too... 🤷